napari/_tests/test_viewer_layer_parity.py

```diff
@@ -127,9 +127,20 @@ def test_signature(layer):
 
     code = inspect.getsource(method)
 
-    args = re.search(rf'layer = layers\.{name}\((.+?)\)', code, flags=re.S)
-    # get the arguments & normalize whitepsace
-    args = ' '.join(args.group(1).split())
+    if name == 'Image':
```

Comment:
I have to say I don't really understand this, or what the original code was doing. Can you maybe add a comment above the if statement about what the whole block is trying to achieve and then maybe make it more clear what is different with our current approach. It doesn't look to me like the actual function signature is changing (in terms of names of things etc.)

Reply:
will add comment.  this test was using `code = inspect.getsource(method)` to grab the source code of the (e.g.) `add_image` method and asserts that somewhere in the method, there _literally_ appears the line `layer = layers.Image(data, rgb=rgb, is_pyramid=is_pyramid, ...)` with every parameter typed out as it is in the corresponding `Layer.__init__` method.  That doesn't work with this approach, which uses `layers.Image(image, **i_kwargs)` to prevent needing to zip up and then unzip 16 different variables.

I think it'd eventually be nicer to eventually rewrite this such that the behavior is tested, rather than a specific implementation, but that's tough to do.